traits/observers/_observer_change_notifier.py

```diff
@@ -0,0 +1,183 @@
+    will be an instance of ``ObserverChangeNotifier`` on the ``CTrait`` for
+    ``bar`` for maintaining the observers for ``baz``, when ``bar`` changes.
+
+    """
+
+    def __init__(
+            self, *, observer_handler, event_factory, prevent_event,
+            graph, handler, target, dispatcher):
+        """
+
+        Parameters
+        ----------
+        observer_handler : callable(event, graph, handler, target, dispatcher)
+            The handler for maintaining observers and notifiers.
+            ``event`` is the object created by the ``event_factory`` and is
+            associated with the change event that triggers this notifier.
+            The rest of the arguments are provided by the arguments to this
+            notifier.
+        event_factory : callable(*args, **kwargs) -> object
+            A factory function for creating the event object to be sent to
+            the observer_handler. The call signature must be compatible with
+            the call signature expected by the observable this notifier is
+            used with. e.g. for CTrait, the call signature will be
+            ``(object, name, old, new)``.
+        prevent_event : callable(event) -> boolean
+            A callable for controlling whether the observer_handler should be
+            invoked. It receives the event created by the event factory and
+            returns true if the event should be prevented, false if the event
+            should be fired.
+        graph : ObserverGraph
+            An object describing what traits are being observed on an instance
+            of ``HasTraits``, e.g. observe mutations on a list referenced by
+            a specific named trait.
+        handler : callable(event)
+            The user handler being maintained when a container object changes.
+            A weak reference is created for the handler if it is an instance
+            method. If the instance method handler is garbage collected, this
+            notifier will be silenced.
+        target : object
+            An object for defining the context of the user's handler notifier.
+            A weak reference is created for the target. If the target is
+            garbage collected, this notifier will be silenced.
+            This is typically an instance of HasTraits acting as a container
+            for an observable on which this notifier is attached to. It would
+            be seen by the user as the "owner" of the observer.
+        dispatcher : callable(function, event)
+            Callable for dispatching the user's handler.
+        """
+        self.observer_handler = observer_handler
+        self.event_factory = event_factory
+        self.prevent_event = prevent_event
+        self.graph = graph
+        self.target = weakref.ref(target)
+        if isinstance(handler, types.MethodType):
+            self.handler = weakref.WeakMethod(handler)
+        else:
+            self.handler = partial(_return, value=handler)
```

Comment:
Perhaps a name better than `_return` which helps explain what's going on, or just use a lambda.  I had to think about what was happening and go searching around for `_return`.

Reply:
I define the function here instead and I will do the same for the other notifier.